flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -83,27 +86,37 @@ default void validate(TableSchema schema) throws ValidationException {}
     }
 
     /**
-     * Quotes the identifier. This is used to put quotes around the identifier in case the column
-     * name is a reserved keyword, or in case it contains characters that require quotes (e.g.
-     * space). Default using double quotes {@code "} to quote.
+     * Quotes the identifier.
+     *
+     * <p>Used to put quotes around the identifier if the column name is a reserved keyword or
+     * contains characters requiring quotes (e.g., space). Default using double quotes {@code "} to
+     * quote.
+     *
+     * @return the quoted identifier.
      */
     default String quoteIdentifier(String identifier) {
         return "\"" + identifier + "\"";
```

Comment:
The default implementation seems broken to me because it doesn't escape the identifier. That being said, should we really offer a default implementation for this?

Reply:
Looks like every dialect already overrides this default 🤦 removing